### PR TITLE
Fix flake8-return warnings

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
## Summary of changes

Fix [flake8-return](https://pypi.org/project/flake8-return/) issues reported by [ruff](https://docs.astral.sh/ruff/rules/#flake8-return-ret) 0.1.9.

### Pull Request Checklist
- [ ] Changes have tests
- [ ] News fragment added in [`newsfragments/`].
  _(See [documentation][PR docs] for details)_

[`newsfragments/`]: https://github.com/pypa/setuptools/tree/master/newsfragments
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

File: pkg_resources/__init__.py [hunk 1]
```diff
@@ -1141,8 +1141,7 @@ def obtain(self, requirement, installer=None):
         None is returned instead.  This method is a hook that allows subclasses
         to attempt other ways of obtaining a distribution before falling back
         to the `installer` argument."""
-        if installer is not None:
-            return installer(requirement)
+        if installer return installer(requirement) else None
```
Comment (abravalheri) [inline pkg_resources/__init__.py:1144]: This is probably one of the things that is causing problems (syntax error).

Comment (DimitriPapadopoulos) [reply pkg_resources/__init__.py]: Fixed.

**Comment by abravalheri** [comment]
Thank you very much @DimitriPapadopoulos .


Should `flake8-return` be added to the ruff configuration as well after these fixes?

**Comment by DimitriPapadopoulos** [comment]
Yes, it could be added. Some rules should be disabled though. I have not fixed such issues:
``` python
if ...:
    return ...
else:
    return ...
```
I find the symmetry makes them more readable than:
``` python
if ...:
    return ...
return ...
```

**Comment by abravalheri** [comment]
Thank you very much @DimitriPapadopoulos, let's merge this for now...
Later we can add the `flake8-return`.

> Some rules should be disabled though.

I am not sure about this. I am more keen to keep the linter configuration as dry as possible, ideally just working out of the box.

**Pull request merged**